sia/miner/mine.go

```diff
@@ -86,7 +86,9 @@ func (m *Miner) SolveBlock() (b consensus.Block, solved bool, err error) {
 	// to find a winnning solution.
 	for maxNonce := b.Nonce + iterations; b.Nonce != maxNonce; b.Nonce++ {
 		if b.CheckTarget(target) {
+			m.mu.RLock()
 			m.blockChan <- b
+			m.mu.RUnlock()
```

Comment:
We added some tests and discovered a race condition.

When you call UpdateMiner() you can change the channel. Here we send a block down a channel without having a lock on the state, which means you could potentially send a block down a channel at the same time that the channel is being updated.